### PR TITLE
Implement built-in http server for miner state reports

### PR DESCRIPTION
**Pull request by jean-m-cyr**
A simple http server.

### CODE REVIEW

File: ethminer/MinerAux.h [hunk 1]
```diff
@@ -391,6 +392,10 @@ class MinerCLI
 		{
 			m_api_port = atoi(argv[++i]);
 		}
+		else if ((arg == "--http-port") && i + 1 < argc)
+		{
+			m_http_port = atoi(argv[++i]);
```
Comment (AndreaLanfranchi) [inline ethminer/MinerAux.h]: I' d not use atoi as it does not throw.
Would prefer stoi and implement a check for ports 0-65535 (positive/negative)

Comment (jean-m-cyr) [reply ethminer/MinerAux.h]: I was lazy and just cut and paste the lines above.

Comment (AndreaLanfranchi) [reply ethminer/MinerAux.h]: Yes happened to me too. :)

There are also a lot of stol to unsigned which should be stoul.
A good reason to begin amend all of them.

Nevertheless I will review positive to allow me testing.
Thank you jean for this.

**Comment by invidtiv** [comment]
Sorry, for this question, how can I pull this to my machine for testing ? linux...

**Pull request merged**